lambda_tiler/handler.py

```diff
@@ -162,64 +260,69 @@ def metadata(
     payload_compression_method="gzip",
     binary_b64encode=True,
 )
-def tile(
+def tile_handler(
     z: int,
     x: int,
     y: int,
     scale: int = 1,
-    ext: str = "png",
+    ext: str = None,
     url: str = None,
     indexes: Union[str, Tuple[int]] = None,
     expr: str = None,
     nodata: Union[str, int, float] = None,
     rescale: str = None,
     color_formula: str = None,
     color_map: str = None,
-):
-    """Handle tile requests."""
-    if ext == "jpg":
-        driver = "jpeg"
-    elif ext == "jp2":
-        driver = "JP2OpenJPEG"
-    else:
-        driver = ext
-
+) -> Tuple[str, str, BinaryIO]:
+    """Handle /tiles requests."""
     if indexes and expr:
         raise TilerError("Cannot pass indexes and expression")
+
     if not url:
         raise TilerError("Missing 'url' parameter")
 
     if isinstance(indexes, str):
         indexes = tuple(int(s) for s in re.findall(r"\d+", indexes))
 
-    tilesize = scale * 256
-
     if nodata is not None:
         nodata = numpy.nan if nodata == "nan" else float(nodata)
 
+    tilesize = scale * 256
     if expr is not None:
-        tile, mask = expression(url, x, y, z, expr=expr, tilesize=tilesize)
+        tile, mask = expression(
+            url, x, y, z, expr=expr, tilesize=tilesize, nodata=nodata
+        )
     else:
-        tile, mask = main.tile(
-            url, x, y, z, indexes=indexes, tilesize=tilesize)
+        tile, mask = cogTiler.tile(
+            url, x, y, z, indexes=indexes, tilesize=tilesize, nodata=nodata
+        )
+
+    if not ext:
+        ext = "jpg" if mask.all() else "png"
```

Comment:
@palmerj I think this should fix what we discussed. basically if no `ext` is given (`/tiles/1/2/3`) the code will choose between jpg or png